src/server/pkg/storage/chunk/reader.go

```diff
@@ -4,109 +4,84 @@ import (
 	"bytes"
 	"compress/gzip"
 	"context"
-	"fmt"
 	"io"
-	"math"
 	"path"
 
-	"github.com/pachyderm/pachyderm/src/client/pkg/grpcutil"
 	"github.com/pachyderm/pachyderm/src/server/pkg/obj"
-	"modernc.org/mathutil"
 )
 
-// ReaderFunc is a callback that returns the next set of data references
-// to a reader.
-type ReaderFunc func() ([]*DataRef, error)
-
-// Reader reads a set of DataRefs from chunk storage.
+// Reader reads data from chunk storage.
 type Reader struct {
-	ctx              context.Context
-	objC             obj.Client
-	dataRefs         []*DataRef
-	curr             *DataRef
-	buf              *bytes.Buffer
-	r                *bytes.Reader
-	len              int64
-	f                ReaderFunc
-	onSplit          func()
-	bytesBeforeSplit int64
+	ctx      context.Context
+	objC     obj.Client
+	dataRefs []*DataRef
+	curr     *DataReader
 }
 
-func newReader(ctx context.Context, objC obj.Client, f ...ReaderFunc) *Reader {
-	r := &Reader{
-		ctx:  ctx,
-		objC: objC,
-		buf:  &bytes.Buffer{},
-	}
-	if len(f) > 0 {
-		r.f = f[0]
+func newReader(ctx context.Context, objC obj.Client, dataRefs ...*DataRef) *Reader {
```

Comment:
Who uses this package and how do they construct `Reader`s? Or would things still compile if `Reader` was made private? If so I think it'd be a good to do that, it doesn't make a lot of sense to me to have a public struct without a public method to construct.